src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/FluidPump.java

```diff
@@ -166,21 +174,21 @@ else if (fluid.getType() == Material.WATER || fluid.getType() == Material.BUBBLE
         }
     }
 
-    private boolean isLiquid(Block block) {
-        if (block.isLiquid()) {
+    private boolean isSource(Block block) {
+        if (isLiquid(block.getType())) {
```

Comment:
Why not Block#isLiquid() ?

Reply:
That only returns true to lava and water. But we also want true for bubble columns.

Reply:
Block#isLiquid() returns true for WATER, LAVA and BUBBLE_COLUMN.
It does the exact same like your method, only that it is a wrapper for NMS code which checks a flag. But that flag is set to true for all three.

Reply:
Well then spigot javadocs aren't complete...

Reply:
To be honest... Does that really surprise you? :LUL: